candle-transformers/src/models/quantized_gemma3.rs

```diff
@@ -133,14 +136,14 @@ struct LayerWeights {
     neg_inf: Tensor,
 
     // Cache
-    kv_cache: Option<(Tensor, Tensor)>,
+    kv_cache: C,
 
     // Tracing
     span_attn: tracing::Span,
     span_mlp: tracing::Span,
 }
 
-impl LayerWeights {
+impl<C: KvCache> LayerWeights<C> {
```

Comment:
Does it need to be generic? It feels like we're expecting a specific behaviour from the cache